File: app/src/main/java/code/name/monkey/retromusic/activities/bugreport/model/Report.java
package code.name.monkey.retromusic.activities.bugreport.model;

import code.name.monkey.retromusic.activities.bugreport.model.github.ExtraInfo;

public class Report {

  private final String description;

  private final DeviceInfo deviceInfo;

  private final ExtraInfo extraInfo;

  private final String title;

  public Report(String title, String description, DeviceInfo deviceInfo, ExtraInfo extraInfo) {
    this.title = title;
    this.description = description;
    this.deviceInfo = deviceInfo;
    this.extraInfo = extraInfo;
  }

  public String getDescription() {
    return description
        + "\n\n"
        + "-\n\n"
        + deviceInfo.toMarkdown()
        + "\n\n"
        + extraInfo.toMarkdown();
  }

  public String getTitle() {
    return title;
  }
}

File: app/src/main/java/code/name/monkey/retromusic/activities/bugreport/model/Report.kt
package code.name.monkey.retromusic.activities.bugreport.model

import code.name.monkey.retromusic.activities.bugreport.model.github.ExtraInfo

class Report(
    val title: String,
    private val description: String,
    private val deviceInfo: DeviceInfo?,
    private val extraInfo: ExtraInfo
) {
    fun getDescription(): String {
        return """
            $description
            
            -
            
            ${deviceInfo?.toMarkdown()}
            
            ${extraInfo.toMarkdown()}
            """.trimIndent()
    }
}